### PR TITLE
fix #18

### LINKED ISSUE #18: Bug: doesn't initialize from an object with null prototype
StackFrame constructor doesn't accept object with null prototype:

```js
> const StackFrame = require('stackframe');
undefined
> new StackFrame({fileName:'lol'})
{ fileName: 'lol' }
> new StackFrame(Object.assign(Object.create(null),{fileName:'lol'}))
{}
> new StackFrame('lol'.match(/(?<fileName>\w+)/).groups)
{}
```

```
$ npm ls stackframe
├── stackframe@1.0.4 
└─┬ stacktrace-gps@3.0.2
  └── stackframe@1.0.4  deduped
```

### PR DESCRIPTION
**Pull request by caub**
Hey @eriwen, could yo have a look?

### CODE REVIEW

**Comment by eriwen** [comment]
Hey @caub. Could you please add a test that ensures #18 stays fixed?

**Comment by caub** [comment]
@eriwen I add this test, but `npm test` isn't working for me, there's probably something to install

```
> gulp test

fs.js:27
const { Math, Object } = primordials;
                         ^

ReferenceError: primordials is not defined
    at fs.js:27:26
    at req_ (/home/caub/dev/stackframe/node_modules/natives/index.js:143:24)
    at Object.req [as require] (/home/caub/dev/stackframe/node_modules/natives/index.js:55:10)
    at Object.<anonymous> (/home/caub/dev/stackframe/node_modules/vinyl-fs/node_modules/graceful-fs/fs.js:1:37)
```

**Comment by niftylettuce** [comment]
It (primordials) wasn't added until Node 10 or something I think. Tests run on 8 right now. Not at desk but can look later.

**Comment by caub** [comment]
I added a fix commit with https://stackoverflow.com/a/55926692/3183756 I'm using node12 and https://github.com/gulpjs/undertaker/issues/54#issuecomment-314082908

(I'm using node12, with this commit it should support all node versions)

File: gulpfile.js [hunk 1]
```diff
@@ -25,44 +25,44 @@ gulp.task('test', function(done) {
...
-});
-
-gulp.task('test-ci', ['copy', 'dist'], function(done) {
-    new karma.Server({
-        configFile: __dirname + '/karma.conf.ci.js',
-        singleRun: true
-    }, done).start();
-});
-
 gulp.task('copy', function() {
```
Comment (caub) [inline gulpfile.js]: moved up, the task must be defined before use

File: gulpfile.js [hunk 2]
```diff
@@ -25,44 +25,44 @@ gulp.task('test', function(done) {
...
     gulp.src(sources)
         .pipe(gulp.dest('dist'));
 });
 
-gulp.task('dist', ['copy'], function() {
+gulp.task('dist', gulp.series(['copy'], function() {
```
Comment (caub) [inline gulpfile.js]: cf https://github.com/gulpjs/undertaker/issues/54#issuecomment-314082908

**Comment by niftylettuce** [comment]
@caub What's left for this PR to do?  Do we need to upgrade to Gulp 4.x and rewrite gulpfile.js?  Or switch to NPM scripts?  If we do it one way or another, we need to make it consistent across the entire org.

**Comment by caub** [comment]
I upgraded gulp to 4.x on this repo to be able to run tests locally, I can left it to 3.x if you prefer

**Comment by niftylettuce** [comment]
@caub 	Could you also upgrade all other repos to 4.x?  Would be immensely helpful.  I know I'm asking a lot, if you don't have the time I could try to squeeze some in this weekend.

**Comment by caub** [comment]
Sorry, I don't want to rewrite anything else to gulp4, I could rather remove gulp on this repo (and use npm scripts), for other repos I think it can be incremental as it's not breaking anythng, it's just a devDep

**Comment by caub** [comment]
So what do you choose?

- undo all gulp changes in this PR
- keep it like this
- drop completely gulp, use npm scripts as it simplifies the flow and gulp is not really helpful

**Comment by niftylettuce** [comment]
npm scripts would be awesome

**Comment by caub** [comment]
@niftylettuce ok, great idea, I'll make another PR for this later

I've removed the gulp4 commit for this PR, to keep it on topic, and merged

**Comment by eriwen** [comment]
Thanks for your contribution, @caub. I now have some time to give stacktrace.js some love and have merged your PR.

**Pull request merged**